pkg/host/internal/infiniband/ib_guid_pool_test.go

```diff
@@ -0,0 +1,105 @@
+package infiniband
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	netlinkLibPkg "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/internal/lib/netlink"
+	netlinkMockPkg "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/internal/lib/netlink/mock"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/internal/network"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/fakefilesystem"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/helpers"
+)
+
+var _ = Describe("ibGUIDPool", func() {
+	var (
+		tempDir        string
+		netlinkLibMock *netlinkMockPkg.MockNetlinkLib
+		testCtrl       *gomock.Controller
+
+		createJsonConfig func(string) string
+	)
+
+	BeforeEach(func() {
+		var err error
+		tempDir, err = os.MkdirTemp("", "ibguidconfig")
+		Expect(err).NotTo(HaveOccurred())
+
+		createJsonConfig = func(content string) string {
+			configPath := "/config.json"
+			helpers.GinkgoConfigureFakeFS(&fakefilesystem.FS{
+				Files: map[string][]byte{configPath: []byte(content)},
+			})
+
+			return configPath
+		}
+
+		testCtrl = gomock.NewController(GinkgoT())
+		netlinkLibMock = netlinkMockPkg.NewMockNetlinkLib(testCtrl)
+		netlinkLibMock.EXPECT().LinkList().Return([]netlinkLibPkg.Link{}, nil).Times(1)
+	})
+
+	AfterEach(func() {
+		os.RemoveAll(tempDir)
+		testCtrl.Finish()
+	})
+	It("should parse correct json config and return a map", func() {
+		vfGuid1, err := ParseGUID("00:00:00:00:00:00:00:00")
+		Expect(err).NotTo(HaveOccurred())
+		vfGuid2, err := ParseGUID("00:00:00:00:00:00:00:01")
+		Expect(err).NotTo(HaveOccurred())
+		rangeStart, err := ParseGUID("00:00:00:00:00:00:01:00")
+		Expect(err).NotTo(HaveOccurred())
+		rangeEnd, err := ParseGUID("00:00:00:00:00:00:01:02")
+		Expect(err).NotTo(HaveOccurred())
+
+		configPath := createJsonConfig(fmt.Sprintf(`[{"pciAddress":"0000:3b:00.0","guids":["%s", "%s"]},{"pciAddress":"0000:3b:00.1","rangeStart":"%s","rangeEnd":"%s"}]`, vfGuid1.String(), vfGuid2.String(), rangeStart.String(), rangeEnd.String()))
+
+		pool, err := newIbGUIDPool(configPath, netlinkLibMock, network.New(nil, nil, nil, nil))
+		Expect(err).NotTo(HaveOccurred())
+
+		guid, err := pool.GetVFGUID("0000:3b:00.0", 0)
```

Comment:
can you add comment what each one is testing ? or better yet use Table style test for this